playground/backend/internal/executors/java_helper.go

```diff
@@ -0,0 +1,186 @@
+	if err != nil {
+		grpclog.Error("Preparation: remove additional package: Error during create new temporary file")
+		return err
+	}
+
+	if err := removePackageString(file, tmp); err != nil {
+		grpclog.Error("Preparation: remove additional package: Error during move data from original file to to temporary file")
+		return err
+	}
+
+	if err := tmp.Close(); err != nil {
+		grpclog.Error("Preparation: remove additional package: Error during Close temporary file")
+		return err
+	}
+	if err := file.Close(); err != nil {
+		grpclog.Error("Preparation: remove additional package: Error during Close original file")
+		return err
+	}
+
+	pathSlice := strings.Split(filePath, "/")
+	fileName := pathSlice[len(pathSlice)-1]
+	if err := os.Rename(tmp.Name(), fileName); err != nil {
+		grpclog.Error("Preparation: remove additional package: Error during rename temporary file")
+		return err
+	}
+	return nil
+}
+
+func removePackageString(reader io.Reader, writer io.Writer) error {
+	reg := regexp.MustCompile(`package ([\w]+\.)+[\w]+;`)
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+		matches := reg.FindAllString(line, -1)
+		for _, str := range matches {
+			line = strings.ReplaceAll(line, str, "")
+		}
+		if _, err := io.WriteString(writer, line+"\n"); err != nil {
+			return err
+		}
+	}
+	return scanner.Err()
+}
+
+func removePublicClassModification(filePath string, args ...interface{}) error {
+	file, err := os.Open(filePath)
+	if err != nil {
+		grpclog.Errorf("Preparation: remove public modification for class: Error during open file: %s", filePath)
+		return err
+	}
+
+	tmp, err := os.Create("temporary.java")
+	if err != nil {
+		grpclog.Error("Preparation: remove public modification for class: Error during create new temporary file")
+		return err
+	}
+
+	if err := replace(file, tmp); err != nil {
+		grpclog.Error("Preparation: remove public modification for class: Error during replace and move data from original file to to temporary file")
+		return err
+	}
+
+	if err := tmp.Close(); err != nil {
+		grpclog.Error("Preparation: remove public modification for class: Error during Close temporary file")
+		return err
+	}
+	if err := file.Close(); err != nil {
+		grpclog.Error("Preparation: remove public modification for class: Error during Close original file")
+		return err
+	}
+
+	pathSlice := strings.Split(filePath, "/")
+	fileName := pathSlice[len(pathSlice)-1]
+	if err := os.Rename(tmp.Name(), fileName); err != nil {
+		grpclog.Error("Preparation: remove public modification for class: Error during rename temporary file")
+		return err
+	}
+	return nil
+}
+
+func replace(reader io.Reader, writer io.Writer) error {
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+		line = strings.ReplaceAll(line, "public class ", "class ")
+		if _, err := io.WriteString(writer, line+"\n"); err != nil {
+			return err
+		}
+	}
+	return scanner.Err()
+}
```

Comment:
Can we also combine this `replace` function with `removePackageString`? I suppose we can provide a regexp as an argument to make a generic function.